more_itertools/more.py

```diff
@@ -180,18 +180,19 @@ def last(iterable, default=_marker):
     raise ``ValueError``.
     """
     try:
-        try:
-            # Try to access the last item directly
+        if isinstance(iterable, Sequence):
```

Comment:
it might be safer and more robust to stick with the try/except approach, in case someone has a strange object that supports `getitem` but not `len`. Might also want to test this case.

Reply:
I don't like the try/except approach, because of the `__getitem__(-1)` ambiguity - do I mean the sequence index or the mapping key? So this was intentional - non-sequences fall back to the `consume all but the last item` behavior.

Reply:
ok sounds good. I considered getting rid of the special case for sequences, because `reversed()` also works on sequences that don't have a `__reversed__` method (and thus removing a special case), but in any case we would have to deal with the 3.8.0 bug first, so the common case would still have to go through a preliminary check anyway.